ldap/servers/slapd/libglobs.c

```diff
@@ -4792,6 +4822,25 @@ config_set_maxdescriptors(const char *attrname, char *value, char *errorbuf, int
         }
     }
 
+#ifdef LINUX
+#ifdef ENABLE_EPOLL
+    maxVal = (int64_t)slapdFrontendConfig->maxuserwatches;
```

Comment:
I don't think we should add an extra setting here. maxuserwatches is always going to have to be at least equal or greater than conntable, meaning that we already have a setting for this as a limit. So I'd say strip out all the maxuserwatches stuff and check that conntable size is =< fs.maxuserwatches and that's it. Then if conntable > fs.maxuserwaches, then clamp conntable to fs.maxuserwatches. 

In general "adding more tunables" is not a good use of time/effort since most tunables are never altered or touched.

Reply:
I agree with the assessment about creating excess tunables. I didn't actually create a new tunable.  max_user_watches is stored in slapdFrontendConfig so that it can be accessed from multiple locations.  At all of the locations where related tunables are configured, they are clamped to the smaller of the specific tunable or max_user_watches.  This seemed the best solution given the number of tunables that clamp the connection table size.

I was concerned that without adding the other clamps it might be confusing for administrators to be denied a tunable change with little or no obvious explanation.

If it is preferable, I could remove all clamps except for the one in config_set_conntablesize.

Reply:
IMHO those checking looks valid to me. Before setting descriptors, conntable-sz, reserveddesc it checks the set values are aligned with max_user_watches, that looks safe and verbose enough to help the admin.